From adobriyan@gmail.com  Mon Aug  6 20:04:41 2007
From: Alexey Dobriyan <adobriyan@gmail.com>
Date: Mon, 6 Aug 2007 23:47:45 +0400
Subject: Fix Off-by-one in /sys/module/*/refcnt
To: Tejun Heo <htejun@gmail.com>
Cc: Kay Sievers <kay.sievers@vrfy.org>, akpm@osdl.org, gregkh@suse.de, linux-kernel@vger.kernel.org
Message-ID: <20070806194745.GB5842@martell.zuzino.mipt.ru>
Content-Disposition: inline


sysfs internals were changed to not pin module in question.

Signed-off-by: Alexey Dobriyan <adobriyan@gmail.com>
Acked-by: Kay Sievers <kay.sievers@vrfy.org>
Acked-by: Tejun Heo <htejun@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 kernel/module.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/kernel/module.c
+++ b/kernel/module.c
@@ -784,8 +784,7 @@ EXPORT_SYMBOL_GPL(symbol_put_addr);
 static ssize_t show_refcnt(struct module_attribute *mattr,
 			   struct module *mod, char *buffer)
 {
-	/* sysfs holds a reference */
-	return sprintf(buffer, "%u\n", module_refcount(mod)-1);
+	return sprintf(buffer, "%u\n", module_refcount(mod));
 }
 
 static struct module_attribute refcnt = {
